pkg/autostart/autostart.go

```diff
@@ -0,0 +1,125 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
```

Comment:
I don't think we should simply return if the file already exists: Maybe the user has re-installed Lima to a different location and wants to update the autostart file so it continues to work.

```suggestion
	if _, err := os.Stat(unitPath); err != nil && !errors.Is(err, os.ErrNotExist) {
		return err
	}
```